internal/services/azurestackhci/stack_hci_logical_network_resource.go

```diff
@@ -0,0 +1,485 @@
+									ValidateFunc: validation.StringMatch(
+										regexp.MustCompile(`^[a-zA-Z0-9][\-\.\_a-zA-Z0-9]{0,78}[a-zA-Z0-9]$`),
+										"name must be between 2 and 80 characters and can only contain alphanumberic characters, hyphen, dot and underline",
+									),
+								},
+
+								"address_prefix": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ForceNew:     true,
+									ValidateFunc: validation.IsCIDR,
+								},
+
+								"next_hop_ip_address": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ForceNew:     true,
+									ValidateFunc: validation.IsIPv4Address,
+								},
+							},
+						},
+					},
+
+					"vlan_id": {
+						Type:         pluginsdk.TypeInt,
+						Optional:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.IntAtLeast(0),
+					},
+				},
+			},
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (StackHCILogicalNetworkResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r StackHCILogicalNetworkResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AzureStackHCI.LogicalNetworks
+
+			var config StackHCILogicalNetworkResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			id := logicalnetworks.NewLogicalNetworkID(subscriptionId, config.ResourceGroupName, config.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			payload := logicalnetworks.LogicalNetworks{
+				Name:     pointer.To(config.Name),
+				Location: location.Normalize(config.Location),
+				Tags:     tags.Expand(config.Tags),
+				ExtendedLocation: &logicalnetworks.ExtendedLocation{
+					Name: pointer.To(config.CustomLocationId),
+					Type: pointer.To(logicalnetworks.ExtendedLocationTypesCustomLocation),
+				},
+				Properties: &logicalnetworks.LogicalNetworkProperties{
+					VMSwitchName: pointer.To(config.VirtualSwitchName),
+					Subnets:      expandStackHCILogicalNetworkSubnet(config.Subnet),
+					DhcpOptions: &logicalnetworks.LogicalNetworkPropertiesDhcpOptions{
+						DnsServers: pointer.To(config.DNSServers),
+					},
+				},
+			}
+
+			future, err := client.CreateOrUpdate(ctx, id, payload)
+			if err != nil {
+				return fmt.Errorf("performing create %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			if err := future.Poller.PollUntilDone(ctx); err != nil {
+				return fmt.Errorf("polling after create %s: %+v", id, err)
+			}
```

Comment:
Why aren't we using the combined `CreateOrUpdateThenPoll` here?

Reply:
changed